cmd/provisioner-localpv/app/provisioner.go

```diff
@@ -125,7 +125,7 @@ func (p *Provisioner) Provision(opts pvController.VolumeOptions) (*v1.Persistent
 	if stgType == "hostpath" {
 		return p.ProvisionHostPath(opts, pvCASConfig)
 	}
-	if stgType == "device" {
+	if stgType == "device" || *opts.PVC.Spec.VolumeMode == v1.PersistentVolumeBlock {
```

Comment:
There should be a error check saying -- not supported if block mode is received with anything other than device mode.